src/drivers/uploadthing.ts

```diff
@@ -0,0 +1,90 @@
+import { defineDriver } from "./utils";
+import { ofetch, $Fetch } from "ofetch";
+import { UTApi } from "uploadthing/server";
+
+export interface UploadThingOptions {
+  apiKey: string;
+}
+
+export default defineDriver<UploadThingOptions>((opts) => {
+  let client: UTApi;
+  let utFetch: $Fetch;
+
+  const getClient = () => {
+    return (client ??= new UTApi({
+      apiKey: opts.apiKey,
+      fetch: ofetch.native,
+    }));
+  };
+
+  const getUTFetch = () => {
+    return (utFetch ??= ofetch.create({
+      baseURL: "https://uploadthing.com/api",
+      headers: {
+        "x-uploadthing-api-key": opts.apiKey,
+      },
+    }));
+  };
+
+  return {
+    hasItem(key) {
+      // This is the best endpoint we got currently...
+      return getUTFetch()("/getFileUrl", {
+        body: { fileKeys: [key] },
+      }).then((res) => res.ok);
+    },
+    getItem(key) {
+      return ofetch(`https://utfs.io/f/${key}`);
+    },
+    getItemRaw(key) {
+      return ofetch
+        .native(`https://utfs.io/f/${key}`)
+        .then((res) => res.arrayBuffer());
+    },
+    getItems(items) {
+      return Promise.all(
+        items.map((item) =>
+          ofetch(`https://utfs.io/f/${item.key}`).then((res) => ({
```

Comment:
If there is no bulk support in API, you can safely remove `getItems`. unstorage (core) will make same parallel call anyway.